### ISSUE TITLE
Different converter - P0ody Files

### ISSUE THREAD

**Issue by Psantium** [tusing/reddit-ffn-bot#81]
Have you ever used http://p0ody-files.com/ff_to_ebook/ as a converter? IMO it works better than FicSave, and takes no time if it has already been converted because it saves the fics it converts. And its formatting has the fix information as well.

Has been implemented but I will try and get in touch with P0ody to see if he can rig it up for FictionPress as well.

/u/StuxCrystal said it didn't work for him, but it works fine for me, /u/Psantium, and /u/tusing.

https://www.reddit.com/r/HPfanfiction/comments/3e57bb/fanfictionbot_usage_tips_updated_check_em_out_here/ctcv546

**Comment by Psantium** [comment]
Okay, I did some tests.

When /u/tusing implemented the p0ody downloader, he was using the link: http://p0ody-files.com/ff_to_ebook/download.php?id={id}&isSplit=0&filetype=epub, of course replacing the {id} with the actual story ID.

**If the fic was not downloaded before, this will not work.** I tested this out with a random fic of the new Harry Potter. I put ID: 11402789 into the previous link, and got [this](http://p0ody-files.com/ff_to_ebook/download.php?id=11402789&isSplit=0&filetype=epub), which says, "Couldn't locate file on database." 

**If it is *FictionPRESS*, this will not work.** I tested out a [fictionpress I'd guess someone would download if they can](https://www.fictionpress.com/s/2780124/1/Rise-From-The-Ashes-OLD), first with the entire link, which didn't work, then with just the ID, which got back a *fanfic* (not fictionpress) with the same ID.

However, if it is a **Fanfiction** ID in the link, and it has been downloaded before, it will work perfectly fine, and will automatically download– **way faster than FicSave**. I tested this with [HPMOR](https://www.fanfiction.net/s/5782108/1/Harry-Potter-and-the-Methods-of-Rationality), which is ID: 5782108. It downloaded automatically in less than a second. 

TL;DR:
* if a fanfic has not been downloaded before, this link: http://p0ody-files.com/ff_to_ebook/download.php?id={id}&isSplit=0&filetype=epub will not work.
* The converter does not support FictionPress
* if the fanfic has been downloaded before, it will automatically download really quickly.

**Comment by tusing** [comment]
Thanks for the info! Stux and I were talking about this earlier.

Unfortunately, I think for now we're stuck with FicSave until we find a way to either better implement poody's or find a faster, better downloader.

**Comment by Psantium** [comment]
I'll try contacting p0ody himself to see if he can make it work better for the bot.

**Comment by StuxSoftware** [comment]
It would help if he would join this discussion.

**Comment by tusing** [comment]
The source would be helpful, at the very least. I'm uncomfortable with the idea of such a useful tool being confined to one server.

**Comment by StuxSoftware** [comment]
I don't think, it would be a good idea to host our own service as it raises the probability of being banned by ffn.

**Comment by p0ody** [comment]
Hey guys, im currently working on a new version of the site to allow fic from more website to be converted, so i'll not add something fancy on this version, ion the meantime, i added a $_GET["id"] handler on the simple version (No ajax), i think it might work properly for you to integrate it in your bot.

Use this url instead.
http://www.p0ody-files.com/ff_to_ebook/mobile/makeEpub.php?id=$ficID

Give me your feedback if it works as expected

**Comment by tusing** [comment]
Thanks, p0ody!

**Comment by tusing** [comment]
@p0ody , is there a way to make your URL also work for FictionPress? Since both sites seem to be the same, implementing FP downloads shouldn't be too difficult.

**Comment by StuxSoftware** [comment]
I closed the issue for now until p0ody has finished his new version. Feel free to reopen when he released the new Downloader.

**Comment by p0ody** [comment]
Here we go guys, it should be working properly (Haven't tested it much though.

New link

http://www.p0ody-files.com/ff_to_ebook/ffn-bot/index.php?id=$ficID&source=$site

$ficID = Id of the fic of course
$site = ff or fp, depending if from fanfiction.net or fictionpress (because the same ID can be used on both site).

Hopes this works for you guys.

**Comment by tusing** [comment]
Thanks, @p0ody! We'll use it in the next major update.

**Comment by p0ody** [comment]
I can also easily add mobi support for your link.

Im at work ATM, i'll look into it once I get home.

**Comment by p0ody** [comment]
Alright, just add &filetype=epub|mobi to the url:
http://www.p0ody-files.com/ff_to_ebook/ffn-bot/index.php?id=$ficID&source=$site&filetype=$epubORmobi

**Comment by tusing** [comment]
Awesome! We'll implement this in 1.3.0
